src/native/mqtt_connection.c

```diff
@@ -672,7 +679,8 @@ jshort JNICALL Java_software_amazon_awssdk_crt_mqtt_MqttClientConnection_mqttCli
     /* pin the topic so it lives as long as the unsub request */
     jni_topic = (*env)->NewGlobalRef(env, jni_topic);
     unsub_ack->jni_objects[0] = jni_topic;
-    struct aws_byte_cursor topic = aws_jni_byte_cursor_from_jstring(env, jni_topic);
+    struct aws_byte_cursor topic = aws_jni_byte_cursor_from_jstring_acquire(env, jni_topic);
+    unsub_ack->strings[0] = (const char*)topic.ptr; /* string will be cleaned up in callback after unsub */
 
     uint16_t msg_id =
         aws_mqtt_client_connection_unsubscribe(connection->client_connection, &topic, s_on_op_complete, unsub_ack);
```

Comment:
do we free the string in error paths?

Reply:
Yes, cleaning up the async_callback will release the string and the pinned reference.